src/com/amazon/ion/benchmark/schema/TypeReference.java

```diff
@@ -0,0 +1,48 @@
+package com.amazon.ion.benchmark.schema;
+
+import com.amazon.ion.IonStruct;
+import com.amazon.ion.IonSymbol;
+import com.amazon.ion.IonValue;
+import com.amazon.ion.system.IonSystemBuilder;
+
+/**
+ * Processing the constraint value <TYPE_REFERENCE>. Here are the constraints that using <TYPE_REFERENCE> as their value:
+ * {type: <TYPE_REFERENCE>, element: <TYPE_REFERENCE>, all_of: [ <TYPE_REFERENCE>... ], one_of: [ <TYPE_REFERENCE>... ], not: <TYPE_REFERENCE>}
+ * <TYPE_REFERENCE> ::=           <TYPE_NAME>
+ *                    | nullable::<TYPE_NAME>
+ *                    |           <TYPE_ALIAS>
+ *                    | nullable::<TYPE_ALIAS>
+ *                    |           <UNNAMED_TYPE_DEFINITION>
+ *                    | nullable::<UNNAMED_TYPE_DEFINITION>
+ *                    |           <IMPORT_TYPE>
+ *                    | nullable::<IMPORT_TYPE>
+ *
+ */
+// TODO: Support processing <TYPE_ALIAS> and <IMPORT_TYPE>
+public class TypeReference {
+    private ReparsedType typeDefinition;
+    private static final String KEYWORD_TYPE = "type";
+
+    /**
+     * Initializing the newly created TypeReference instance.
+     * @param field represents constraint value <TYPE_REFERENCE>.
+     */
+    public TypeReference(IonValue field) {
```

Comment:
It looks like a `TypeReference` instances are never needed; they're always used only to get the underlying type definition as a `ReparsedType`. Therefore, is this class needed? Or, should the logic in the constructor simply be pulled into a static method?

Reply:
> It looks like a `TypeReference` instances are never needed; they're always used only to get the underlying type definition as a `ReparsedType`. Therefore, is this class needed? Or, should the logic in the constructor simply be pulled into a static method?

It would be a better idea to pull the logic into a static method. In the following commit, the method `parseTypeReference` will be added to the class `IonSchemaUtilities`.